src/schemes/fluid/surface_tension.jl

```diff
@@ -0,0 +1,193 @@
+    (; surface_tension_coefficient) = surface_tension
+    return cohesion_force_akinci(surface_tension, support_radius, mb, pos_diff,
+                                 distance) .- (surface_tension_coefficient * (na - nb))
+end
+
+@fastpow @inline function cohesion_force_akinci(surface_tension::AkinciTypeSurfaceTension,
+                                                support_radius, mb, pos_diff, distance)
+    (; surface_tension_coefficient) = surface_tension
+
+    # Eq. 2
+    # we only reach this function when distance > eps
+    C = 0
+    if distance <= support_radius
+        if distance > 0.5 * support_radius
+            # attractive force
+            C = (support_radius - distance)^3 * distance^3
+        else
+            # distance < 0.5 * support_radius
+            # repulsive force
+            C = 2 * (support_radius - distance)^3 * distance^3 - support_radius^6 / 64.0
+        end
+        C *= 32.0 / (pi * support_radius^9)
+    end
+
+    # Eq. 1 in acceleration form
+    cohesion_force = -surface_tension_coefficient * mb * C * pos_diff / distance
+
+    return cohesion_force
+end
+
+@inline function adhesion_force_akinci(surface_tension::AkinciTypeSurfaceTension,
+                                       support_radius, mb, pos_diff, distance,
+                                       adhesion_coefficient)
+    # Eq. 7
+    # we only reach this function when distance > eps
+    A = 0
+    if distance < support_radius
+        if distance > 0.5 * support_radius
+            A = 0.007 / support_radius^3.25 *
+                (-4 * distance^2 / support_radius + 6 * distance - 2 * support_radius)^0.25
+        end
+    end
+
+    # Eq. 6 in acceleration form with mb being the boundary mass calculated as mb=rho_0 * volume (Akinci boundary condition treatment)
+    adhesion_force = -adhesion_coefficient * mb * A * pos_diff / distance
+
+    return adhesion_force
+end
+
+# section 2.2 in Akinci et al. 2013 "Versatile Surface Tension and Adhesion for SPH Fluids"
+# Note: most of the time this only leads to an approximation of the surface normal
+function calc_normal_akinci(surface_tension::SurfaceTensionAkinci, u_system,
+                            v_neighbor_system, u_neighbor_system,
+                            neighborhood_search, system, neighbor_system::FluidSystem)
+    (; smoothing_kernel, smoothing_length, cache) = system
+
+    @threaded for particle in each_moving_particle(system)
+        particle_coords = current_coords(u_system, system, particle)
+
+        for neighbor in eachneighbor(particle_coords, neighborhood_search)
+            neighbor_coords = current_coords(u_system, system, neighbor)
+
+            pos_diff = particle_coords - neighbor_coords
+            distance = norm(pos_diff)
+            # correctness strongly depends on this being a symmetric distribution of particles
+            if sqrt(eps()) < distance <= compact_support(smoothing_kernel, smoothing_length)
```

Comment:
```suggestion
            distance2 = dot(pos_diff)
            # correctness strongly depends on this being a symmetric distribution of particles
            if eps() < distance2 <= compact_support(smoothing_kernel, smoothing_length)^2
            distance = sqrt(distance2)
```

maybe a small performance gain?